src/fomo.jl

```diff
@@ -0,0 +1,399 @@
+export fomo, FomoSolver, FoSolver, R2, tr_step, r2_step
+
+abstract type AbstractFirstOrderSolver <: AbstractOptimizationSolver end
+
+abstract type AbstractFomoMethod end
```

Comment:
Should this be called `AbstractFOMethod` since it does not depend on momentum?

Reply:
Yes, It should